adding yarn gen to Cont.md (#1144)
temporalio/documentation: CONTRIBUTING.md

yarn format
```

To reformat incremental changes:

```
yarn prettier --write .prettierrc.json
```

## Guide generation

> The [Concepts](https://docs.temporal.io/docs/temporal/#) and [Application development](https://docs.temporal.io/docs/application-development-guide) guides are generated through the [index.js](guide-gen/index.js) script.

Reviewer: Good call.

Maybe add something about the guides and the guide generation tooling to be something new and in a sort of alpha state.

And add that the guides should not be added to directly.

Add to source files first, then run `yarn gen` -

> The [Concepts](https://docs.temporal.io/docs/temporal/#) and [Application development](https://docs.temporal.io/docs/application-development-guide) guides are generated through the [index.js](guide-gen/index.js) script.

Reviewer: Good call.

Maybe add something about the guides and the guide generation tooling to be something new and in a sort of alpha state.

And add that the guides should not be added to directly.

Add to source files first, then run `yarn gen` -

yarn format
```

To reformat incremental changes:

```
yarn prettier --write .prettierrc.json
```

## Guide generation

The guide generation script is an Alpha state.

> Each json file in [guide-gen/guide-configs](guide-gen/guide-configs) releates to a specific concept that's expressed by a declartive configuiration reperesented by conglomerate of related markdown files. These configuration files generate the [Concepts](https://docs.temporal.io/docs/temporal/#) and [Application development](https://docs.temporal.io/docs/application-development-guide) guides by means of the [index.js](guide-gen/index.js) script.

Reviewer 2: ```suggestion
Each JSON configuration file in [guide-gen/guide-configs](guide-gen/guide-configs) relates to a specific concept and identifies related Markdown files. The [index.js](guide-gen/index.js) script uses these configuration files to generate guides such as [Concepts](https://docs.temporal.io/docs/temporal/#) and [Application development](https://docs.temporal.io/docs/application-development-guide).
```
I generalized the guides because we likely will have more.

> The guide generation script is an Alpha state.
> 
> Each json file in [guide-gen/guide-configs](guide-gen/guide-configs) releates to a specific concept that's expressed by a declartive configuiration reperesented by conglomerate of related markdown files. These configuration files generate the [Concepts](https://docs.temporal.io/docs/temporal/#) and [Application development](https://docs.temporal.io/docs/application-development-guide) guides by means of the [index.js](guide-gen/index.js) script.
> 
> To update the Concept or Application development guide, make your changes in the source file. For example, if you found a typo on the Concept section _What is a Task_, you would open [docs/concepts/what-is-a-task.md](docs/concepts/what-is-a-task.md) and make the edit directly there. Then rebuild the Concept guide by running the following commands:
> 
> ```bash
> yarn
> ```
> 
> ```bash
> yarn gen
> ```
> 
> To build and view the site locally run:
> 
> ```bash
> yarn start
> ```
> 

Reviewer: ```suggestion
Many of the guides in the docs folder are generated from a script.
Do not edit these generated guides directly (a commented warning is at the top of each one), edit the source files and run the following command to re-generate the guides with the source file changes.

```bash
yarn gen
```

The [guide generation script](guide-gen/index.js) is an Alpha state.

Each json file in [guide-gen/guide-configs](guide-gen/guide-configs) corresponds to a guide.
```

This would be my preference for now.
